Error running generator app: Error: Cannot find module 'yeoman-environment/transform'

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

I am using a fork of JHipster which is up to date with main.
When I try to run jhipster I get an error, stacktrace shown below.
If I switch to tag v7.3.1 then the error does not occur.

```
INFO! Using JHipster version installed globally
INFO! Executing import-jdl model.jdl
INFO! The JDL is being parsed.
warn: An Entity name 'User' was used: 'User' is an entity created by default by JHipster. All relationships toward it will be kept but any attributes and relationships from it will be disregarded.
INFO! Found entities: Department, Employee, Roster.
INFO! The JDL has been successfully parsed
INFO! Generating 1 application.
Error running generator app: Error: Cannot find module 'yeoman-environment/transform'
Require stack:
- E:\jhipster\generator-jhipster\generators\bootstrap\index.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\util\esm.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\environment.js
- E:\jhipster\generator-jhipster\cli\environment-builder.js
- E:\jhipster\generator-jhipster\cli\program.js
- E:\jhipster\generator-jhipster\cli\cli.js
- E:\jhipster\generator-jhipster\cli\jhipster.js
Error: Cannot find module 'yeoman-environment/transform'
Require stack:
- E:\jhipster\generator-jhipster\generators\bootstrap\index.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\util\esm.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\environment.js
- E:\jhipster\generator-jhipster\cli\environment-builder.js
- E:\jhipster\generator-jhipster\cli\program.js
- E:\jhipster\generator-jhipster\cli\cli.js
- E:\jhipster\generator-jhipster\cli\jhipster.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (E:\jhipster\generator-jhipster\generators\bootstrap\index.js:28:5)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at requireOrImport (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\util\esm.js:14:14)
    at Object.get [as jhipster:bootstrap] (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js:49:16)
    at Store.get (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js:72:44)
    at Environment.composability.getByNamespace (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\namespace-composability.js:21:37)
    at Environment.get (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\environment.js:621:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'E:\\jhipster\\generator-jhipster\\generators\\bootstrap\\index.js',
    'E:\\jhipster\\generator-jhipster\\node_modules\\yeoman-environment\\lib\\util\\esm.js',
    'E:\\jhipster\\generator-jhipster\\node_modules\\yeoman-environment\\lib\\store.js',
    'E:\\jhipster\\generator-jhipster\\node_modules\\yeoman-environment\\lib\\environment.js',
    'E:\\jhipster\\generator-jhipster\\cli\\environment-builder.js',
    'E:\\jhipster\\generator-jhipster\\cli\\program.js',
    'E:\\jhipster\\generator-jhipster\\cli\\cli.js',
    'E:\\jhipster\\generator-jhipster\\cli\\jhipster.js'
  ]
}
ERROR! Cannot find module 'yeoman-environment/transform'
Require stack:
- E:\jhipster\generator-jhipster\generators\bootstrap\index.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\util\esm.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\environment.js
- E:\jhipster\generator-jhipster\cli\environment-builder.js
- E:\jhipster\generator-jhipster\cli\program.js
- E:\jhipster\generator-jhipster\cli\cli.js
- E:\jhipster\generator-jhipster\cli\jhipster.js
Error: Cannot find module 'yeoman-environment/transform'
Require stack:
- E:\jhipster\generator-jhipster\generators\bootstrap\index.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\util\esm.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js
- E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\environment.js
- E:\jhipster\generator-jhipster\cli\environment-builder.js
- E:\jhipster\generator-jhipster\cli\program.js
- E:\jhipster\generator-jhipster\cli\cli.js
- E:\jhipster\generator-jhipster\cli\jhipster.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (E:\jhipster\generator-jhipster\generators\bootstrap\index.js:28:5)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at requireOrImport (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\util\esm.js:14:14)
    at Object.get [as jhipster:bootstrap] (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js:49:16)
    at Store.get (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\store.js:72:44)
    at Environment.composability.getByNamespace (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\namespace-composability.js:21:37)
    at Environment.get (E:\jhipster\generator-jhipster\node_modules\yeoman-environment\lib\environment.js:621:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'E:\\jhipster\\generator-jhipster\\generators\\bootstrap\\index.js',
    'E:\\jhipster\\generator-jhipster\\node_modules\\yeoman-environment\\lib\\util\\esm.js',
    'E:\\jhipster\\generator-jhipster\\node_modules\\yeoman-environment\\lib\\store.js',
    'E:\\jhipster\\generator-jhipster\\node_modules\\yeoman-environment\\lib\\environment.js',
    'E:\\jhipster\\generator-jhipster\\cli\\environment-builder.js',
    'E:\\jhipster\\generator-jhipster\\cli\\program.js',
    'E:\\jhipster\\generator-jhipster\\cli\\cli.js',
    'E:\\jhipster\\generator-jhipster\\cli\\jhipster.js'
  ]
}
```

##### **Motivation for or Use Case**

I can't use jhipster, this will need to be solved for the next version.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->
clone jhipster
npm link
try to run jhipster in a new project dir

##### **Related issues**

None found

##### **Suggest a Fix**

Please solve the missing dependency

##### **JHipster Version(s)**

> 3.7.1 ... Fork of main at commit ea922e7

##### **JHipster configuration**

It doesn't run, it crashes

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

None, just trying to run jhipster on command line 

##### **Browsers and Operating System**

Windows 10 Prod + chrome

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -221,9 +221,11 @@ Go to the [generator-jhipster project](https://github.com/jhipster/generator-jhi
 
 ### Set NPM to use the cloned project
 
-In your cloned `generator-jhipster` project, type `npm link`.
+In your cloned `generator-jhipster` project, run `npm link` and then run `npm ci`.
 
-This will do a symbolic link from the global `node_modules` version to point to this folder, so when we run `jhipster`, you will now use the development version of JHipster.
+`npm link` will make a symbolic link from the global `node_modules` version to point to this folder, so when we run `jhipster`, you will now use the development version of JHipster.
+
+`npm ci` will do a clean install of all of the project dependencies.
 
 For testing, you will want to generate an application, and there is a specific issue here: for each application, JHipster installs a local version of itself. This is made to enable several applications to each use a specific JHipster version (application A uses JHipster 3.1.0, and application B uses JHipster 3.2.0).
 

